### PR TITLE
Replace 'global' with 'classFilter' callback

### LINKED ISSUE #20: [Feature Request] Annotation @NotObfuscate and configuration to match package / class
`@NotObfuscate` would be very useful with `global = true`.

Configuration like the following will be even more convenient:
```
lsparanoid {
  obfuscateOnly 'com.example1.**'
  neverObfuscate 'com.example2.**'
}
```

Please consider if possible.

### PR DESCRIPTION
**Pull request by mirfatif**
Resolves #20

### CODE REVIEW

File: gradle-plugin/src/main/java/org/lsposed/lsparanoid/plugin/LSParanoidExtension.kt [hunk 1]
```diff
@@ -21,7 +21,9 @@ import com.android.build.api.variant.Variant
 
 open class LSParanoidExtension {
     var seed: Int? = null
+    @Deprecated("Use 'classFilter: (String) -> Boolean'.")
```
Comment (yujincheng08) [inline gradle-plugin/src/main/java/org/lsposed/lsparanoid/plugin/LSParanoidExtension.kt:24]: remove it and bump ver

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: gradle-plugin/src/main/java/org/lsposed/lsparanoid/plugin/LSParanoidExtension.kt (hunk 1 above)
Comment (mirfatif) [reply gradle-plugin/src/main/java/org/lsposed/lsparanoid/plugin/LSParanoidExtension.kt:24]: > remove it and bump ver

Should I remove the `@Deprecated` annotation or the `global` property?

Comment (yujincheng08) [reply gradle-plugin/src/main/java/org/lsposed/lsparanoid/plugin/LSParanoidExtension.kt:24]: both

**Comment by mirfatif** [comment]
@yujincheng08 any plan to publish `0.6.0` to maven repo?

**Comment by yujincheng08** [comment]
this pr breaks configuration cache

**Comment by vvb2060** [comment]
revert!

**Pull request merged**